src/main/java/com/hotels/beeju/core/BeejuCore.java

```diff
@@ -65,48 +76,93 @@ public BeejuCore() {
   public BeejuCore(String databaseName) {
     this(databaseName, Collections.emptyMap());
   }
-  
+
   public BeejuCore(String databaseName, HiveConf preConfiguration, HiveConf postConfiguration) {
     this(databaseName, convertToMap(preConfiguration), convertToMap(postConfiguration));
   }
-  
+
   public BeejuCore(String databaseName, Map<String, String> preConfiguration) {
     this(databaseName, preConfiguration, Collections.emptyMap());
   }
-  
+
   public BeejuCore(String databaseName, Map<String, String> preConfiguration, Map<String, String> postConfiguration) {
     checkNotNull(databaseName, "databaseName is required");
     this.databaseName = databaseName;
     configure(preConfiguration);
-    
+
+    configureFolders();
+
     driverClassName = EmbeddedDriver.class.getName();
     conf.setBoolean("hcatalog.hive.client.cache.disabled", true);
     connectionURL = "jdbc:derby:memory:" + UUID.randomUUID() + ";create=true";
     conf.setVar(HiveConf.ConfVars.METASTORECONNECTURLKEY, connectionURL);
     conf.setVar(HiveConf.ConfVars.METASTORE_CONNECTION_DRIVER, driverClassName);
     conf.setVar(HiveConf.ConfVars.METASTORE_CONNECTION_USER_NAME, METASTORE_DB_USER);
     conf.setVar(HiveConf.ConfVars.METASTOREPWD, METASTORE_DB_PASSWORD);
+    conf.setVar(HiveConf.ConfVars.METASTORE_CONNECTION_POOLING_TYPE, "NONE");
     conf.setBoolVar(HiveConf.ConfVars.HMSHANDLERFORCERELOADCONF, true);
+
     // Hive 2.x compatibility
     conf.setBoolean("datanucleus.schema.autoCreateAll", true);
     conf.setBoolean("hive.metastore.schema.verification", false);
+
     // override default port as some of our test environments claim it is in use.
-    conf.setInt("hive.server2.webui.port", 0); // ConfVars.HIVE_SERVER2_WEBUI_PORT
-    
-    try {
-      // overriding default derby log path to go to tmp
-      String derbyLog = File.createTempFile("derby", ".log").getCanonicalPath();
-      System.setProperty("derby.stream.error.file", derbyLog);
+    conf.setIntVar(HiveConf.ConfVars.HIVE_SERVER2_WEBUI_PORT, 0);
```

Comment:
Can we initialise this with some other port number rather than 0 (which disables it outright)?

Reply:
My understanding was that with "0" it chooses a random port, if I set this to a specific number I got some issue with tests complaining about the port already being used (as the comment above describes). Either way, I haven't changed this value from what it was before I just changed the variable name from a string to type so I'd rather leave it.

Reply:
According to the doc, setting it to 0 or a negative number disables the Web UI feature. I had to use a different (fixed) port to make it work, although it may be something specific to Hive 3.

Reply:
Ah, OK, well, it seems to have worked here so I suggest we leave it and change it in your Hive 3 PR.